sound/soc/sof/intel/hda-dai.c

```diff
@@ -452,9 +458,24 @@ int sdw_hda_dai_hw_params(struct snd_pcm_substream *substream,
 	if (!hext_stream)
 		return -ENODEV;
 
-	/* in the case of SoundWire we need to program the PCMSyCM registers */
+	/*
+	 * in the case of SoundWire we need to program the PCMSyCM registers. In case
+	 * of aggregated devices, we need to define the channel mask for each sublink
+	 * by reconstructing the split done in soc-pcm.c
+	 */
+	for_each_rtd_cpu_dais(rtd, cpu_dai_id, dai) {
+		if (dai == cpu_dai)
```

Comment:
cpu_dai_id is not initialized and if cpu_dai can't match up with dai in this loop, cpu_dai_id is uncertain

Reply:
@RanderWang `cpu_dai_id` is used as a loop iterator
```
#define for_each_rtd_cpu_dais(rtd, i, dai)				\
	for ((i) = 0;							\
	     ((i) < rtd->dai_link->num_cpus) && ((dai) = asoc_rtd_to_cpu(rtd, i)); \
	     (i)++)
```

Reply:
@RanderWang has a point, there are no existing precedents of using the loop iterator outside the loop for this macro, and it could very well be wrong. Using a 'found' boolean would ensure that the default value of zero is not used. will fix.

Reply:
fixed.